### PR TITLE
fix: Improvements in cert and pre-reqs scripts

### PR DESCRIPTION
**Pull request by rjaini**
Improvements in cert and pre-reqs scripts

### CODE REVIEW

File: registry/Scripts/pre-reqs.ps1 [hunk 1]
```diff
@@ -43,23 +43,25 @@ function New-ResourceGroup (
     [string] $ResourceGroupName
 )
 {
-    Write-Host "Check if resource group($ResourceGroupName) already exist" 
+    Write-Host "Check if resource group ($ResourceGroupName) already exist"
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: I think it is `exists`

File: registry/Scripts/pre-reqs.ps1 [hunk 2]
```diff
@@ -160,31 +164,33 @@ function New-StorageAccount (
                     -StorageAccountBlobContainer "images"
 #>
 function New-StorageAccountContainer (
-    [Parameter(Mandatory = $true, HelpMessage = "Name of the resource group under which stroage account to be created.")]
+    [Parameter(Mandatory = $true, HelpMessage = "Name of the resource group under which storage blob to be created.")]
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: I would say that `The storage account's resource group` is good enough

File: registry/Scripts/pre-reqs.ps1 [hunk 3]
```diff
@@ -227,19 +233,24 @@ function New-KeyVault (
 {
     Write-Host "Check if key vault($KeyVaultName) exist."
     $ErrorActionPreference = "SilentlyContinue";
-    $keyVaultDetails = Get-AzureRmKeyVault -Name $KeyVaultName
+    $keyVaultDetails = Get-AzureRmKeyVault -Name $KeyVaultName -ResourceGroupName $ResourceGroupName 
     $ErrorActionPreference = "Continue"; #Turning errors back on
     if (-not $keyVaultDetails)
     {
-        Write-Host "Creating key vault($KeyVaultName) as it does not exist."
+        Write-Host "Creating key vault ($KeyVaultName) as it does not exist."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: I think `Key Vault` is more appropriate

File: registry/Scripts/pre-reqs.ps1 [hunk 4]
```diff
@@ -227,19 +233,24 @@ function New-KeyVault (
...
                                                -Sku $Sku `
                                                -EnabledForDeployment
+        $ErrorActionPreference = "Continue"; #Turning errors back on
+        if (-not $keyVaultDetails){
+            throw "Creation of KeyVault ($KeyVaultName) failed."
+        }
     }
     else {        
-        Write-Host "Key-vault($KeyVaultName) already exist. Skipping creation."
+        Write-Host "Key-vault ($KeyVaultName) already exist. Skipping creation."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: extra `-`

File: registry/Scripts/pre-reqs.ps1 [hunk 5]
```diff
@@ -302,9 +313,15 @@ function New-KeyVaultSecret (
                                 -Name $SecretName `
                                 -SecretValue $secureSecret `
                                 -ContentType $ContentType | Out-Null
+        $ErrorActionPreference = "SilentlyContinue";
+        $keyVaultSecretDetails = Get-AzureKeyVaultSecret -VaultName $KeyVaultName -Name $SecretName
+        $ErrorActionPreference = "Continue"; #Turning errors back on
+        if (-not $keyVaultSecretDetails){
+            throw "Creation of KeyVault secret ($SecretName) failed. Check if specified key vault already exist under a subscription."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: `Key Vault`

Comment (jadarsie) [reply registry/Scripts/pre-reqs.ps1]: `if the specified`

File: registry/Scripts/pre-reqs.ps1 [hunk 6]
```diff
@@ -562,28 +578,28 @@ function Set-ContainerRegistryPrerequisites
                                 -StorageAccountName $StorageAccountName `
                                 -StorageAccountBlobContainer $StorageAccountBlobContainer
 
-    Write-Host "Checking if on storage account($StorageAccountName), ServicePrincipleId($ServicePrincipleId) already has access."
+    Write-Host "Checking if on storage account ($StorageAccountName), ServicePrincipleId ($ServicePrincipleId) already has access."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: `on` probably a typo

File: registry/Scripts/pre-reqs.ps1 [hunk 7]
```diff
@@ -640,11 +656,17 @@ function Get-VMImageSku (
...
-        $skuDetails | Select-Object Skus
+        Write-Host "Available Skus:"
+        $skuDetails | ForEach-Object {
+            '{0}' -f $_.Skus
+          }
+        #$skuDetails | Select-Object Skus
     }
     else {
-        Write-Host "No image sku install with Publisher as $PublisherName and offer as $Offer)."
+        Write-Host "No image sku install with Publisher ($PublisherName) and offer ($Offer)."
```
Comment (jadarsie) [inline registry/Scripts/pre-reqs.ps1]: `SKU`
install?

File: registry/Scripts/self-signed.ps1 [hunk 8]
```diff
@@ -15,6 +15,9 @@
 .Parameter CertificateFileExportPath
   Certificate file export path including certificate filename.
 
+.Parameter GenerateWildCardCert
+  Flag to generate wild card cert.
```
Comment (jadarsie) [inline registry/Scripts/self-signed.ps1]: `wildcard`
https://en.wikipedia.org/wiki/Wildcard_DNS_record

File: registry/Scripts/pre-reqs.ps1 (hunk 1 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: Yup. updated.

File: registry/Scripts/pre-reqs.ps1 (hunk 2 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: updated.

File: registry/Scripts/pre-reqs.ps1 (hunk 4 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: updated.

File: registry/Scripts/pre-reqs.ps1 (hunk 6 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: updated.

File: registry/Scripts/pre-reqs.ps1 (hunk 7 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: Updated.

File: registry/Scripts/pre-reqs.ps1 (hunk 5 above)
Comment (rjaini) [reply registry/Scripts/pre-reqs.ps1]: Updated.

**Pull request merged**